### ISSUE TITLE
Selecting view type or filter should not shift focus to the top of the page

### ISSUE THREAD

**Issue by alundgard** [sul-dlss/exhibits#2473]
### Current behavior

Selecting a view type or filter using the keyboard or screen reader shifts the focus back to the top of the page.

### Steps to reproduce

Example page: [Spotlight Accessibility for SODA - San Francisco Buildings](https://exhibits.stanford.edu/a11y-for-soda/browse/san-francisco-buildings).

Navigate to the view type buttons with the keyboard or screen reader and make a selection. Also, navigate to the sidebar filters and make a selection.

### Possible solution

WCAG Level A: [Understanding Success Criterion 3.2.1: On Focus](https://www.w3.org/WAI/WCAG21/Understanding/on-focus.html).

SODA recommends moving the focus to the beginning of the updated content (in this case, the exhibit items). Doing so would mean jumping over the Search input field, which sits between the view options and exhibit items. Or, when selecting a filter from the sidebar, focus would jump across sections of the page.

Another possibility may be to leave focus in its current position (at the view type button or sidebar filter), but with an indication that it has been selected (see #2472).

### Reported by SODA
> Violation: Ensure content updates define focus updates appropriately
> Severity: Major
>
> [Issue]
> When activating the buttons for list vs. grid view, the focus shifts to the top of the page rather than being on the new view of content.
> 
> [User Impact]
> When content within a page changes without reloading the page and without appropriate focus changes, assistive technologies such as screen readers may not be aware of the changes or the user may only be alerted that a change in content occurred and have to review the entire page to discover exactly what changed.
> 
> [Recommendation]
> When the context of the page changes such as when the user initiates a change that updates a large portion of the page (e.g. in a single page application) focus should move to the beginning of the updated content. Similarly, when a button is activated to open a dialog, focus should be moved into the dialog. When the dialog is closed, focus needs to return to the element that triggered the dialog.
> 
> When activating any filter, the focus should either remain on the control or move to the search results. Use Javascript .focus() or another technique to manage focus. 
> 
> https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/focus

**Comment by jcoyne** [comment]
The different view types are actually different web pages.

**Comment by alundgard** [comment]
> The different view types are actually different web pages.
Thanks, I see that selecting filters from the sidebar has the same issue.

**Comment by dnoneill** [comment]
@alundgard @jcoyne clicking on those icons actually reloads the page and goes it the equivalent of a new url. Does that change the recommendation or is it the same?

**Comment by jcoyne** [comment]
Yes. That's what I've been trying to say.

**Comment by alundgard** [comment]
Because each facet/view setting loads a new page, it seems that it's not possible to easily solve this focus issue.

Unless you have other ideas, we may need to rely on the [skip links](https://webaim.org/techniques/skipnav/) (Skip to main content, Skip to search, Skip to first result) to help users to jump to the relevant section of the page.

These skip links are present in the Spotlight DOM, but they don't appear when navigating by keyboard tabbing the way they do in other Blacklight applications (Archives, EarthWorks).
<img width="1497" alt="skiplinks-archives" src="https://github.com/user-attachments/assets/e0cabd87-b792-4ca0-ace4-a291e7ac7d85">
<img width="1497" alt="skiplinks-earthworks" src="https://github.com/user-attachments/assets/f42b30a4-f4bc-43c7-aad5-2ac054f67257">
Absent other approaches, my suggestion for handling this issue is to improve the way skip links appear in Spotlight.

**Comment by hudajkhan** [comment]
Below is what these skip link options look like in various Blacklight-based systems including a modified version of Exhibits. @alundgard it would be good to know which option would make sense for Exhibits. 

First, here is a screenshot from a modified version of Exhibits on the search results page:

<img width="465" alt="Image" src="https://github.com/user-attachments/assets/ebdca64d-e8f9-4884-be4f-593989a50570" />
<img width="370" alt="Image" src="https://github.com/user-attachments/assets/a38c1f03-9e0f-4d76-b189-5206809cdfd9" />

These are screenshots from Spotlight:
<img width="529" alt="Image" src="https://github.com/user-attachments/assets/2d5b1293-5eeb-40c9-ab6a-e0dbcc33ddd0" />


This is a screenshot from EarthWorks:
<img width="623" alt="Image" src="https://github.com/user-attachments/assets/b063c7a2-430b-42cd-87ce-927b3ded2b5f" />

These are screenshots from SearchWorks:
<img width="198" alt="Image" src="https://github.com/user-attachments/assets/e883343e-74f0-4a25-93f9-9480a2335ab6" />

**Comment by alundgard** [comment]
@hudajkhan Thanks for the screenshots.
- I think the EarthWorks design pattern is straightforward. Let's try to emulate that pattern.
- There are only 2 skip links on the home page because there are no search results on that page. If you visit a [search result page](https://earthworks.stanford.edu/?search_field=all_fields&q=), the `Skip to first result` link will appear in the link list.
- I wonder if this work intersects with [this PR](https://github.com/sul-dlss/exhibits/pull/2751) from @jcoyne ?

**Comment by hudajkhan** [comment]
Yes https://github.com/sul-dlss/exhibits/pull/2750 and https://github.com/sul-dlss/exhibits/pull/2751 are both related. The result looks like

![Image](https://github.com/user-attachments/assets/64f67f84-32cc-4e11-b078-2712e4c1feeb)
